code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -303,10 +303,15 @@
 		if(new_xeno)
 			qdel(new_xeno)
 		return
-	if(tier != XENO_TIER_ONE || !regression )
-		new_xeno.upgrade_xeno(upgrade, TRUE)
-	if(!regression && upgrade != XENO_UPGRADE_INVALID)
-		new_xeno.upgrade_xeno(new_xeno.upgrade_prev(), TRUE)
+
+	if(!regression)
+		if(upgrade != XENO_UPGRADE_INVALID)
+			new_xeno.upgrade_xeno(upgrade_prev(), TRUE)
+	else
+		if(tier == XENO_TIER_ONE)
+			new_xeno.upgrade_xeno(XENO_UPGRADE_INVALID, TRUE)
```

Comment:
Why are you setting an invalid upgrade here

Reply:
> Why are you setting an invalid upgrade here

Because of XENO_TIER_ZERO (hivemind, larva) has not upgrade points.